registry/kubernetes/registry.go

```diff
@@ -0,0 +1,351 @@
+package kubernetes
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+import (
+	gxnet "github.com/dubbogo/gost/net"
+	perrors "github.com/pkg/errors"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/registry"
+	"github.com/apache/dubbo-go/remoting/kubernetes"
+)
+
+var (
+	processID = ""
+	localIP   = ""
+)
+
+const (
+	Name              = "kubernetes"
+	RegistryConnDelay = 3
+)
+
+func init() {
+	processID = fmt.Sprintf("%d", os.Getpid())
+	localIP, _ = gxnet.GetLocalIP()
+	extension.SetRegistry(Name, newKubernetesRegistry)
+}
+
+type kubernetesRegistry struct {
+	*common.URL
+	birth int64 // time of file birth, seconds since Epoch; 0 if unknown
+
+	cltLock  sync.Mutex
+	client   *kubernetes.Client
+	services map[string]common.URL // service name + protocol -> service config
+
+	listenerLock   sync.Mutex
+	listener       *kubernetes.EventListener
+	dataListener   *dataListener
+	configListener *configurationListener
+
+	wg   sync.WaitGroup // wg+done for kubernetes client restart
+	done chan struct{}
+}
+
+func (r *kubernetesRegistry) Client() *kubernetes.Client {
+	return r.client
+}
+func (r *kubernetesRegistry) SetClient(client *kubernetes.Client) {
+	r.client = client
+}
+func (r *kubernetesRegistry) ClientLock() *sync.Mutex {
+	return &r.cltLock
+}
+func (r *kubernetesRegistry) WaitGroup() *sync.WaitGroup {
+	return &r.wg
+}
+func (r *kubernetesRegistry) GetDone() chan struct{} {
+	return r.done
+}
+func (r *kubernetesRegistry) RestartCallBack() bool {
+
+	services := []common.URL{}
+	for _, confIf := range r.services {
+		services = append(services, confIf)
+	}
+
+	flag := true
+	for _, confIf := range services {
+		err := r.Register(confIf)
+		if err != nil {
+			logger.Errorf("(kubernetesProviderRegistry)register(conf{%#v}) = error{%#v}",
+				confIf, perrors.WithStack(err))
+			flag = false
```

Comment:
It seems that you can `return false` directly and `return true` at the end of this method. So the variable `flag` could be removed

Reply:
Yes, it would be more straightforward to return the bool value directly。